frontend/packages/console-dynamic-plugin-sdk/src/extensions/user-preferences.ts

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { Extension, ExtensionDeclaration, CodeRef } from '../types';
+
+export enum UserPreferenceFieldType {
+  dropdown = 'dropdown',
+  checkbox = 'checkbox',
+  custom = 'custom',
+}
+
+export type UserPreferenceDropdownField = {
+  type: UserPreferenceFieldType.dropdown;
+  userSettingsKey: string;
+  defaultValue?: string;
+  options: {
+    value: string;
+    label: string;
+  }[];
+};
+
+export type UserPreferenceCheckboxField = {
+  type: UserPreferenceFieldType.checkbox;
+  userSettingsKey: string;
+  label: string;
+  trueValue: string;
+  falseValue: string;
```

Comment:
I don't think this must be strings. It's maybe any(thing) which could be defined in JSON. But at least we should start with `string | number | boolean`. Wdyt?

```suggestion
  trueValue: string | number | boolean;
  falseValue: string | number | boolean;
```

Reply:
yes. we can have non string values for true and false values. Updated types.